protocol/communities_messenger_signers_test.go

```diff
@@ -204,3 +207,221 @@ func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdateSigner() {
+	s.Require().NoError(err)
+
+	// set john as contract owner
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.john.identity.PublicKey))
+	s.collectiblesServiceMock.SetMockCollectibleContractData(chainID, tokenAddress,
+		&communitytokens.CollectibleContractData{TotalSupply: &bigint.BigInt{}})
+
+	hasTokenPermission := func(r *MessengerResponse) bool {
+		return len(r.Communities()) > 0 && r.Communities()[0].HasTokenPermissions()
+	}
+
+	// bob received owner permissions
+	_, err = WaitOnSignaledMessengerResponse(
+		s.bob,
+		hasTokenPermission,
+		"no communities with token permission for Bob",
+	)
+	s.Require().NoError(err)
+
+	// alice received owner permissions
+	_, err = WaitOnSignaledMessengerResponse(
+		s.alice,
+		hasTokenPermission,
+		"no communities with token permission for Alice",
+	)
+	s.Require().NoError(err)
+
+	// simulate Alice received owner token
+	s.collectiblesServiceMock.SetSignerPubkeyForCommunity(community.ID(), common.PubkeyToHex(&s.alice.identity.PublicKey))
+
+	// after receiving owner token - set up control node, set up owner role, kick all members
+	// and request kicked members to rejoin
+	_, err = s.alice.CommunityNewOwnershipKickAllAndRequestRejoin(community.ID())
+	s.Require().NoError(err)
+	community, err = s.alice.communitiesManager.GetByID(community.ID())
+	s.Require().NoError(err)
+	s.Require().True(community.IsControlNode())
+	s.Require().True(common.IsPubKeyEqual(community.ControlNode(), &s.alice.identity.PublicKey))
+	s.Require().True(community.IsOwner())
+
+	// check that client received kick event
+	// Bob will receive request to share RevealedAddresses and send request to join to the control node
+	_, err = WaitOnSignaledMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && !r.Communities()[0].HasMember(&s.bob.identity.PublicKey)
+		},
+		"Bob was not kicked from the community",
+	)
+	s.Require().NoError(err)
+
+	// check that client received kick event
+	// John will receive request to share RevealedAddresses and send request to join to the control node
+	_, err = WaitOnSignaledMessengerResponse(
+		s.john,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && !r.Communities()[0].HasMember(&s.john.identity.PublicKey)
+		},
+		"John was not kicked from the community",
+	)
+	s.Require().NoError(err)
+
+	// Alice auto-accept requests to join with RevealedAddresses
+	_, err = WaitOnMessengerResponse(
+		s.alice,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && len(r.Communities()[0].Members()) == 2
+		},
+		"no community update with accepted request",
+	)
+	s.Require().NoError(err)
+
+	_, err = WaitOnMessengerResponse(
+		s.bob,
+		func(r *MessengerResponse) bool {
+			return len(r.Communities()) > 0 && r.Communities()[0].HasMember(&s.bob.identity.PublicKey)
+		},
+		"Bob was auto-accepted",
+	)
+	s.Require().NoError(err)
```

Comment:
We could add some more checks here to make sure everything is right, like getting the community from Alice's DB and confirming that everyone is a member and that we have Bob's address (John's address will be missing sadly, we need to fix it in another PR)

Reply:
done